### PR TITLE
Python Solution: Out of Place and Diamond Collecter

### PR DESCRIPTION
**Pull request by ryanchou-dev**
All test cases pass for this solution.

### CODE REVIEW

**Review by HarryWangATX — CHANGES_REQUESTED** [review]
prob just add a time complexity

**Comment by ryanchou-dev** [comment]
got it, adding it now.

**Review by HarryWangATX — CHANGES_REQUESTED** [review]
use latex 👀

**Comment by ryanchou-dev** [comment]
oops

**Comment by ryanchou-dev** [comment]
I don't use LaTeX too often, is my formatting correct?

File: solutions/usaco-785.mdx [hunk 1]
```diff
@@ -2,16 +2,41 @@
...
 source: USACO Bronze 2018 January
 title: Out of Place
-author: Maggie Liu
+author: Maggie Liu, Ryan Chou
 ---
 
 [Official Analysis](http://www.usaco.org/current/data/sol_outofplace_bronze_jan18.html)
 
+**Time Complexity:** $\mathcal{O}(N)$
+## Implementation
```
Comment (maggie-j-liu) [inline solutions/usaco-785.mdx:11]: implementation should go above time complexity

File: solutions/usaco-785.mdx [hunk 2]
```diff
@@ -2,16 +2,41 @@
...
+<PySection>
+	
+```py
+fin = open('outofplace.in', 'r')
+
+N = int(fin.readline())
+
+order = [int(fin.readline()) for _ in range(N)]
+
+ordsort = sorted(order)
```
Comment (maggie-j-liu) [inline solutions/usaco-785.mdx:24]:
```suggestion
fin = open('outofplace.in', 'r')
N = int(fin.readline())

order = [int(fin.readline()) for _ in range(N)]
ordsort = sorted(order)
```
idk, i feel like some of these blank lines are unnecessary, but up to you

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: solutions/usaco-785.mdx (hunk 1 above)
Comment (ryanchou-dev) [reply solutions/usaco-785.mdx:11]: alright, thanks!

**Comment by ryanchou-dev** [comment]
anything else I forgot to add?

**Review by maggie-j-liu — APPROVED** [review]

**Comment by ryanchou-dev** [comment]
Added solution for Diamond Collector.

**Comment by ryanchou-dev** [comment]
Is there anything wrong with the Diamond Collector solution?

File: solutions/usaco-639.mdx [hunk 3]
```diff
@@ -50,4 +50,32 @@ int main(){
 
 </CPPSection>
 
+<PySection>
+
+```py
```
Comment (maggie-j-liu) [inline solutions/usaco-639.mdx:55]: use tabs instead of spaces to indent

File: solutions/usaco-639.mdx [hunk 4]
```diff
@@ -50,4 +50,32 @@ int main(){
 
 </CPPSection>
 
+<PySection>
```
Comment (maggie-j-liu) [inline solutions/usaco-639.mdx:53]: also add your name to the authors list

Comment (ryanchou-dev) [reply solutions/usaco-639.mdx:53]: alright, adding it now.

File: solutions/usaco-639.mdx [hunk 5]
```diff
@@ -50,4 +50,32 @@ int main(){
...
+
+for x in range(N):
+    for y in range(N):
+        if diamonds[y] >= diamonds[x] and diamonds[y] <= diamonds[x] + K:
+            curlargest += 1
+    largest = max(largest, curlargest)
+    curlargest = 0
+
+with open('diamond.out', 'w') as fout:
+    fout.write(str(largest) + '\n')
```
Comment (SansPapyrus683) [inline solutions/usaco-639.mdx:76]: just saying, you could make this more concise by writing
```py
print(largest, file=open('diamond.out', 'w'))
```

File: solutions/usaco-639.mdx [hunk 6]
```diff
@@ -50,4 +50,32 @@ int main(){
...
+
+N, K = map(int, fin.readline().split())
+
+diamonds = []
+
+for _ in range(N):
+    diamonds.append(int(fin.readline()))
+
+tracker = []
+largest, curlargest = 0
```
Comment (SansPapyrus683) [inline solutions/usaco-639.mdx:66]: this should error?

File: solutions/usaco-785.mdx [hunk 7]
```diff
@@ -2,16 +2,40 @@
...
+
+swaps = 0
+
+for i in range(N):
+	if order[i] != ordsort[i]:
+		swaps += 1
+
+fout = open('outofplace.out', 'w')
+fout.write(str(swaps - 1))
+fout.close()
```
Comment (SansPapyrus683) [inline solutions/usaco-785.mdx:33]: again, you could write
```py
print(swaps - 1, file=open('outofplace.out', 'w'))
```
instead

File: solutions/usaco-785.mdx [hunk 8]
```diff
@@ -2,16 +2,40 @@
...
+	
+```py
+fin = open('outofplace.in', 'r')
+N = int(fin.readline())
+
+order = [int(fin.readline()) for _ in range(N)]
+ordsort = sorted(order)
+
+swaps = 0
+
```
Comment (SansPapyrus683) [inline solutions/usaco-785.mdx:26]:
```suggestion
```

File: solutions/usaco-639.mdx [hunk 9]
```diff
@@ -50,4 +50,32 @@ int main(){
...
+
```
Comment (SansPapyrus683) [inline solutions/usaco-639.mdx:67]:
```suggestion
```

**Review by SansPapyrus683 — APPROVED** [review]
hi

File: solutions/usaco-639.mdx (hunk 5 above)
Comment (ryanchou-dev) [reply solutions/usaco-639.mdx]: ohhhh ok, adding it.

File: solutions/usaco-639.mdx (hunk 6 above)
Comment (ryanchou-dev) [reply solutions/usaco-639.mdx]: i just realized that earlier, it's fixed now.

File: solutions/usaco-639.mdx (hunk 5 above)
Comment (ryanchou-dev) [reply solutions/usaco-639.mdx]: done. thanks for the suggestion, i'll start using that now. :)

File: solutions/usaco-785.mdx (hunk 7 above)
Comment (ryanchou-dev) [reply solutions/usaco-785.mdx:32]: fixing it.

**Comment by ryanchou-dev** [comment]
lmk if there's anything else.

File: solutions/usaco-639.mdx [hunk 10]
```diff
@@ -50,4 +50,30 @@ int main(){
...
+N, K = map(int, fin.readline().split())
+
+diamonds = []
+
+for _ in range(N):
+	diamonds.append(int(fin.readline()))
+
+tracker = []
+largest, curlargest = 0, 0
```
Comment (SansPapyrus683) [inline solutions/usaco-639.mdx:66]:
```suggestion
largest = 0
curlargest = 0
```
honestly up to you though

**Review by SansPapyrus683 — COMMENTED** [review]
yeah it's poggers

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/usaco-639.mdx (hunk 10 above)
Comment (ryanchou-dev) [reply solutions/usaco-639.mdx:66]: ig that does make it more clear. 👍

**Review by maggie-j-liu — APPROVED** [review]

File: solutions/usaco-639.mdx [hunk 11]
```diff
@@ -50,4 +50,31 @@ int main(){
...
+fin = open('diamond.in', 'r')
+
+N, K = map(int, fin.readline().split())
+
+diamonds = []
+
+for _ in range(N):
+	diamonds.append(int(fin.readline()))
+
+tracker = []
```
Comment (maggie-j-liu) [inline solutions/usaco-639.mdx:65]:
```suggestion
```
this isn't used anywhere, did you forget to remove?

File: solutions/usaco-639.mdx [hunk 12]
```diff
@@ -2,7 +2,7 @@
 id: usaco-639
 source: USACO Bronze 2016 Open
 title: Diamond Collector
-author: Danh Ta Chi Thanh
+author: Danh Ta Chi Thanh, Ryan Chou
 ---
 
 [Official Analysis (Java)](http://www.usaco.org/current/data/sol_diamond_bronze_open16.html)
```
Comment (maggie-j-liu) [inline solutions/usaco-639.mdx:8]: also add time complexity

Comment (ryanchou-dev) [reply solutions/usaco-639.mdx:8]: 👍

File: solutions/usaco-639.mdx (hunk 11 above)
Comment (ryanchou-dev) [reply solutions/usaco-639.mdx]: oops, sorry about that.

File: solutions/usaco-639.mdx [hunk 13]
```diff
@@ -2,13 +2,15 @@
...
 
 ## Implementation
 
+Time Complexity: $\mathcal{O}(n^2)$
```
Comment (maggie-j-liu) [inline solutions/usaco-639.mdx:12]:
```suggestion
Time Complexity: $\mathcal{O}(N^2)$
```

**Review by maggie-j-liu — APPROVED** [review]

File: solutions/usaco-639.mdx [hunk 14]
```diff
@@ -2,13 +2,15 @@
...
 title: Diamond Collector
-author: Danh Ta Chi Thanh
+author: Danh Ta Chi Thanh, Ryan Chou
 ---
 
 [Official Analysis (Java)](http://www.usaco.org/current/data/sol_diamond_bronze_open16.html)
 
 ## Implementation
 
+Time Complexity: $\mathcal{O}(N^2)$
```
Comment (dongliu0426) [inline solutions/usaco-639.mdx:12]:
```suggestion
**Time Complexity:** $\mathcal{O}(N^2)$
```

**Review by dongliu0426 — APPROVED** [review]

**Review by HarryWangATX — APPROVED** [review]

**Pull request merged**